### ISSUE TITLE
AWS.Logger.NLog Getting  ResourceNotFoundException- The specified log stream does not exist

### ISSUE THREAD

**Issue by antonylineesh1** [aws/aws-logging-dotnet#134]
## Description
I am using NLog to logging to cloudwatch from my application, this issue happens when I have cleared some log streams so it keeps throwing exception

**Exception: Amazon.CloudWatchLogs.Model.ResourceNotFoundException: The specified log stream does not exist**  and it is writing  without any limit  in Nlog internal log file. I can get it working by stop & restart the application  that time it will create a new logstream but I don't think that is a proper solution, what my expectation is to create a log stream when it doesn't exist any work around for this.

### Logs
 ```
Exception: Amazon.CloudWatchLogs.Model.ResourceNotFoundException: The specified log stream does not exist.
 ---> Amazon.Runtime.Internal.HttpErrorResponseException: Exception of type 'Amazon.Runtime.Internal.HttpErrorResponseException' was thrown.
   at Amazon.Runtime.HttpWebRequestMessage.GetResponseAsync(CancellationToken cancellationToken)
   at Amazon.Runtime.Internal.HttpHandler`1.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.Unmarshaller.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.ErrorHandler.InvokeAsync[T](IExecutionContext executionContext)
   --- End of inner exception stack trace ---
   at Amazon.Runtime.Internal.HttpErrorResponseExceptionHandler.HandleException(IExecutionContext executionContext, HttpErrorResponseException exception)
   at Amazon.Runtime.Internal.ErrorHandler.ProcessException(IExecutionContext executionContext, Exception exception)
   at Amazon.Runtime.Internal.ErrorHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.EndpointDiscoveryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.EndpointDiscoveryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CredentialsRetriever.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.RetryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.RetryHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.CallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.ErrorCallbackHandler.InvokeAsync[T](IExecutionContext executionContext)
   at Amazon.Runtime.Internal.MetricsHandler.InvokeAsync[T](IExecutionContext executionContext)
   at AWS.Logger.Core.AWSLoggerCore.SendMessages(CancellationToken token)
   at AWS.Logger.Core.AWSLoggerCore.Monitor(CancellationToken token)
```

### Environment

* Build Version: AWS.Logger.NLog 1.5.2 , NLog 4.7.2,NLog.Web.AspNetCore 4.9.2
* OS Info: Windows 10 
* Build Environment: Visual Studio 2019
* Targeted .NET Platform:  .NET Core 3.1

## Resolution
- [ ] :wave: I can/would-like-to implement a fix for this problem myself

---

This is a :bug: bug-report

**Comment by snakefoot** [comment]
Created pull-request:  #135

**Comment by normj** [comment]
The PR is released in version 2.0.1 of AWS.Logger.Core. All of the other packages were version bumped to pick up latest version of AWS.Logger.Core.
